packages/gamut/src/Form/styles/shared-system-props.tsx

```diff
@@ -0,0 +1,125 @@
+import { system, theme } from '@codecademy/gamut-styles';
+import { StandardPropertiesHyphen } from 'csstype';
+
+export type conditionalStyleProps = {
+  error?: boolean;
+  activated?: boolean;
+  isFocused?: boolean | null;
+  isDisabled?: boolean | null;
+};
+
+type iconPaddingProps = {
+  icon?: boolean;
+};
+
+export type conditionalInputStyleProps = conditionalStyleProps &
+  iconPaddingProps;
+
+const transitionConcatenator = (
+  arrayOfProperties: Array<keyof StandardPropertiesHyphen>,
+  transition: string
+) => {
+  const cssString = `${arrayOfProperties.join(
+    ` ${transition},`
+  )} ${transition}`;
+
+  return {
+    transition: cssString,
+  };
+};
+
+export const formBaseStyles = {
+  fontWeight: 'base',
+  fontSize: 16,
+} as const;
+
+export const formBaseComponentStyles = {
+  ...formBaseStyles,
+  width: 1,
+  outline: 'none',
+  bg: 'background',
+  color: `${theme.colors.text}`,
+  minWidth: 'auto',
+} as const;
+
+export const formFieldFocusStyles = {
+  borderColor: 'primary',
+  boxShadow: `inset 0 0 0 1px ${theme.colors.primary}`,
+} as const;
+
+const formFieldDisabledStyles = {
+  bg: 'backgroundEmphasized',
+  borderColor: 'textMuted',
+  color: 'textMuted',
```

Comment:
I really gotta fix the JSX types for this, not being able to use color is super annoying.

```suggestion
  textColor: 'textMuted',
```